crates/runc-shim/src/cgroup_memory.rs

```diff
@@ -0,0 +1,128 @@
+use containerd_shim::{
+    error::{Error, Result},
+    io_error, other_error,
+};
+use std::path::Path;
+use tokio::{
+    fs::{self, File},
+    io::{AsyncBufReadExt, AsyncReadExt, BufReader},
+    sync::mpsc::{self, Receiver},
+};
+
+use nix::sys::eventfd;
+use std::os::unix::io::{AsRawFd, FromRawFd};
+
+pub async fn get_path_from_cgorup(pid: u32) -> Result<String> {
+    let proc_path = format!("/proc/{}/cgroup", pid);
+    let file = File::open(proc_path.clone())
+        .await
+        .map_err(io_error!(e, "open {}.", proc_path))?;
+    let mut lines = BufReader::new(file).lines();
+    while let Some(line) =
+        lines
+            .next_line()
+            .await
+            .map_err(io_error!(e, "read file {} failed.", proc_path))?
+    {
+        if line.contains("memory") {
+            let parts: Vec<&str> = line.split(':').collect();
+            if let Some(path) = parts.last() {
```

Comment:
I believe there is no need to allocate `Vec` here?
Just `line.split(':').last()` should use iterator directly?

Reply:
Ok.